src/palace/manager/service/google_drive/google_drive.py

```diff
@@ -22,22 +22,25 @@
 
 class GoogleDriveService(LoggerMixin):
     def __init__(
-        self,
-        service_account_info: dict[str, Any],
+        self, credentials: Any | None, http: HttpMockSequence | None = None
     ) -> None:
 
-        scopes = ["https://www.googleapis.com/auth/drive"]
-        credentials = service_account.Credentials.from_service_account_info(
-            info=service_account_info, scopes=scopes
-        )
+        if credentials:
+            assert not http
+        if http:
+            assert not credentials
 
-        self.service = build("drive", "v3", credentials=credentials)
+        self.service = build(
+            "drive", "v3", credentials=credentials, http=http  # type:ignore[arg-type]
```

Comment:
Minor: might be nice to have a few less type ignores here. I managed to get them all removed in my other PR.